core/templates/dev/head/components/profileLink.js

```diff
@@ -0,0 +1,74 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directives for creating text and image links to a user's 
+ * profile page.
+ *
+ * @author raine@stanford.edu (raine Hoover)
+ */
+
+oppia.directive('profileLinkText', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      username: '@'
+    },
+    templateUrl: 'components/profileLinkText',
+    controller: ['$scope', function($scope) {
+      $scope.isUsernameLinkable = function(username){
+        return GLOBALS.SYSTEM_USERNAMES.indexOf(username) == -1;
+      };
+    }]
+  };
+}]);
+
+oppia.directive('profileLinkImage', [function() { 
+  return {
+    restrict: 'E',
+    scope: {
+      username: '@',
+      classes: '@',
+      alt: '@'
+    },
+    templateUrl: 'components/profileLinkImage',
+    controller: ['$scope', '$http', '$q', function($scope, $http, $q) {
+      $scope.isUsernameLinkable = function(username){
+        return GLOBALS.SYSTEM_USERNAMES.indexOf(username) == -1;
+      };
+
+      $scope.profileImageURL = '/preferenceshandler/profile_picture_by_username/' + $scope.username;
+      var DEFAULT_PROFILE_IMAGE_PATH = '/images/avatar/user_blue_72px.png';
+      $scope.profilePicture = DEFAULT_PROFILE_IMAGE_PATH;
+      // Returns a promise for the user profile picture, or the default image if
+      // user is not logged in or has not uploaded a profile picture, or the
+      // player is in preview mode.
+      function getUserProfileImage(profileImageURL) {
```

Comment:
We tend to follow the pattern

    var getUserProfileImage = function(profileImageUrl) {
      ...
    };

(Note the naming of the function as an explicit variable, and the casing of 'url' in profileImageUrl.)